script.js

```diff
@@ -1,4 +1,160 @@
+var player1Numbers = [];
+var player2Numbers = [];
+
+// create a function that stores both player's roll and outputs them
+var generatePlayerRoll = function () {
+  var allPlayerRoll = [];
+
+  for (var counter = 0; counter < 2; counter += 1) {
+    var playerRoll = [];
+
+    var diceRoll = rollDice();
+    playerRoll.push(diceRoll);
+
+    // ensure second roll is not the same as the first roll
+    diceRoll = rollDice();
+    while (diceRoll == playerRoll[0]) {
+      diceRoll = rollDice();
+    }
+    playerRoll.push(diceRoll);
+
+    allPlayerRoll.push(playerRoll);
+  }
+
+  player1Roll = allPlayerRoll[0];
+  player2Roll = allPlayerRoll[1];
+};
+
+// create a function that takes in user input on which dice to go first and then outputs the concatenated number
+var concatenateDiceRoll = function (playerInput, playerRoll) {
+  var finalNumber = String(playerRoll[0]) + String(playerRoll[1]);
+
+  if (playerInput == "2") {
+    finalNumber = String(playerRoll[1]) + String(playerRoll[0]);
+  }
+
+  return finalNumber;
+};
+
+// include input validation
+var validateInput = function (playerInput) {
+  var validInput = "";
+
+  if (playerInput == 1 || playerInput == 2) {
+    validInput = true;
+  } else {
+    validInput = false;
+  }
+
+  return validInput;
+};
+
+// create a function that compares player 1 and player 2 numbers and outputs winner
+
+var comparePlayerRolls = function (player1Number, player2Number) {
+  if (player1Number > player2Number) {
+    result = `Player 1: ${player1Number} > Player 2: ${player2Number}<br>
+    Player 1 wins!`;
+    player1Win += 1;
+  } else if (player1Number < player2Number) {
+    result = `Player 2: ${player2Number} > Player 1: ${player1Number}<br>
+    Player 2 wins!`;
+    player2Win += 1;
+  } else {
+    result = `Both players rolled ${player1Number}.<br>
+    It's a draw!`;
+  }
+  return result;
+};
+
+// create a function for dice rolling
+var rollDice = function () {
+  var diceNumber = Math.floor(Math.random() * 6 + 1);
+  return diceNumber;
+};
+
+var showRollOutcome = function (player, dice1Roll, dice2Roll) {
+  var message = `${player}'s Dice Roll<br>
+  *********************** <br>
+  🎲 1 : [${dice1Roll}] <br>
+  🎲 2 : [${dice2Roll}] <br><br>
+  Pick the dice to be ordered first.<br>
+  Enter 1 for Dice 1 and 2 for Dice 2.`;
+  return message;
+};
+
+// scoreboard
+
+var generateScoreboard = function () {
+  var scoreboard = `Leaderboard of ${totalGames} Games: <br><br>
+  Player 1's Wins: ${player1Win} <br>
+  Player 1's Numbers: ${player1Numbers} <br><br>
+  Player 2's Wins: ${player2Win} <br>
+  Player 2's Numbers: ${player2Numbers}
+  `;
+  return scoreboard;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var gameMessage = "";
+
+  if (currentGameStatus == "player 1 turn") {
+    // roll the dice for both players
+    totalGames += 1;
+    generatePlayerRoll();
+    player1dice1 = player1Roll[0];
+    player1dice2 = player1Roll[1];
```

Comment:
This could be refactored into a helper function. Remember that we ideally only want to indicate the flow of the game in the main function. The logic should ideally be carried out in helper functions.